Apple Silicon compatibility added

OnkarBedekar opened the pull request:
This update ensures the project builds and runs correctly on Apple Silicon Macs. Adjusted architecture-specific settings and updated dependencies as needed.

mthomas00 commented on setup/setup_r.R:
```diff
@@ -4,20 +4,70 @@
+    install.packages(CRAN_packages, 
+                     repos = repos,
+                     dependencies = TRUE, 
+                     lib = '../lib/r',
```
When running this script, I get an error that `'lib = "../lib/r"' is not writable`. I think changing the path to "lib/r" works. Update in several other places.

mthomas00 commented on setup/setup_r.R:
```diff
@@ -4,20 +4,70 @@
+    "reticulate", "lfe", "lightgbm", "sentimentr", "scales", "plyr", "tm", 
+    "SnowballC", "textshape", "snakecase", "tidytext"
+  )
+  
```
Add `CRAN_packages <- NULL` because all of these packages install from conda.

OnkarBedekar commented:
I've fixed the R library path in setup_r.R and clarified the package installation source.
Changes have been pushed to my branch.

mthomas00 commented:
Hi Onkar,

Thank you for submitting this. I have worked with it a bit and it appears to work using both Rosetta 2 and using native Apple Silicon. As far as I can tell, this works. (I'm not sure why I got errors when I tried to load `r-brms` this way, for example.). However, I have several comments, mostly on keeping the documentation tidy:

* When creating a pull request, please add a comment linking it to the the issue that it closes. In this case "PR for #7"

* I would like the `README.md` at the top of the project to provide complete instructions for how to install everything required to support the project. Currently, it breaks for me at `R CMD BATCH --no-save ./setup/setup_r.R ./setup/setup_r.Rout`. I have added some further comments on this, but you should make sure the instructions are complete and correct and do not break.

* Please add to `conda_env.yaml` the additional R packages posted [here](https://github.com/mthomas00/template/issues/7#issuecomment-2973146157). I believe they all load into the conda environment, but you should double check.

* Trouble shooting guides should go to the wiki on GitHub. I'm not sure I actually found any.


### Things I'm unsure about:
I'm happy to discuss these if you have a strategy in mind that I don't yet see.

* The content of changes.txt should go in the commit comments or on the issue page if they really took place.

* I'm not sure all of the documentation is necessary or referrs to something real:
   - In `changes.txt`, `Added troubleshooting steps for LaTeX/LyX issues in the documentation.` -- I don't see mention of this any where else in the documentation. 
   - `APPLE_SILICON_MIGRATION.MD` mentions a "Troubleshooting guide" which I don't see. Is this document necessary?
   - In `README.md`, the section `Apple Silicon (M1/M2) Setup` seems extraneous.
   - Generally, using ChatGPT is fine, but I don't want extraneous material added.

* What is the point of `setup/setup_apple_silicon.sh`? All the packages seem to load using conda and those that don't should handled in `setup/setup_R`. Does it do something extra? If not, remove it.

* There are files that appear at the top level: `setup_r.Rout`, `texput.log`. If they are not required please remove them.

Please make these changes, discuss with me where necessary and resubmit the PR and I will review again.

OnkarBedekar commented:
Created these files while experimenting with several setup and troubleshooting methods. I will clean up and publish a streamlined, cleaner version for the final project setup.

OnkarBedekar commented:
Thank you for the thorough review and clear feedback!
I’ve addressed all the points as requested:

- Streamlined the setup and documentation for reproducibility.
- Ensured all R dependencies are managed via conda.
- Removed outdated files and redundant docs.
- Verified a clean setup from scratch—everything works as expected.

Please let me know if there’s anything else you'd like me to adjust or clarify!
